osgp/protocol-adapter-oslp/osgp-adapter-protocol-oslp-elster/src/main/java/org/opensmartgridplatform/adapter/protocol/oslp/elster/infra/messaging/processors/CommonRebootRequestMessageProcessor.java

```diff
@@ -77,6 +77,22 @@ public void processSignedOslpEnvelope(final String deviceIdentification,
         final int retryCount = unsignedOslpEnvelopeDto.getRetryCount();
         final boolean isScheduled = unsignedOslpEnvelopeDto.isScheduled();
 
+        final DeviceResponseHandler deviceResponseHandler = this.createSetRebootDeviceResponseHandler(domain,
+                domainVersion, messageType, retryCount, isScheduled);
+
+        final DeviceRequest deviceRequest = new DeviceRequest(organisationIdentification, deviceIdentification,
+                correlationUid, messagePriority);
+
+        try {
+            this.deviceService.doSetReboot(oslpEnvelope, deviceRequest, deviceResponseHandler, ipAddress);
+        } catch (final IOException e) {
+            this.handleError(e, correlationUid, organisationIdentification, deviceIdentification, domain, domainVersion,
+                    messageType, messagePriority, retryCount);
+        }
+    }
+
+    protected DeviceResponseHandler createSetRebootDeviceResponseHandler(final String domain,
+            final String domainVersion, final String messageType, final int retryCount, final boolean isScheduled) {
         final DeviceResponseHandler deviceResponseHandler = new DeviceResponseHandler() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "deviceResponseHandler". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1488)

Reply:
I agree with Jenkins